### DISCUSSION TITLE
Why doesn't my placeholder work on columns with relationships?

### DISCUSSION THREAD

**Discussion by bsamworthdev** [filamentphp/filament#10202] [Help] [answered]
I've set up a relationship between Customer and Account models.

```
class Customer extends Model
{
    public function account(): HasOne
    {
        return $this->hasOne(Account::class, 'contactid');
    }
}

class Account extends Model
{
    public function individual(): BelongsTo
    {
        return $this->belongsTo(Individual::class, 'contactid');
    }
}
```

So when I create an infolist with the section below the correct field values are loaded. The problem is that when the values are null the placeholder does not show for account.name.

```
Section::make('Customer Details')
                     ->schema([
                         TextEntry::make('account.name')
                             ->label('Account ID')
                             ->placeholder('empty'),  //This never shows
                         TextEntry::make('fullname')
                             ->label('Account Type'), //This shows ok
)]
```

This behaviour seems to happen for all TextEntry fields that use relationships for their values.

I've even checked dd($this->customer->account) and I can see that the name is null.

**Comment by leandrocfe** [comment]
maybe in this case the value is "" instead of null. You could try it
```php
->formatStateUsing(fn ($state): string => filled($state) ? $state : 'empty')
```

**Reply by bsamworthdev** [reply]
Looks like you're right- It is "" instead of null. I can see "empty" appearing in the fields now. However it needs to be a placeholder rather than a value. 

I have tried this: 

`->formatStateUsing(fn ($state) => filled($state) ? $state : null)`

But the field still stays blank.

**Reply by leandrocfe** [reply]
Try `getStateUsing` or `state`
```php
->getStateUsing(fn ($state) => filled($state) ? $state : null)
```

**Comment by bsamworthdev** [comment]
> Try `getStateUsing` or `state`
> 
> ```
> ->getStateUsing(fn ($state) => filled($state) ? $state : null)
> ```

That seems to cause an infinite loop so the page doesn't load at all.

**Reply by leandrocfe** [reply, marked as answer]
Sorry
```php
->getStateUsing(fn ($record) => $record->account->name ?? null)
```

**Comment by bsamworthdev** [comment]
> Sorry
> 
> ```
> ->getStateUsing(fn ($record) => $record->account->name ?? null)
> ```

Cracked it! Thanks
